arma-rs/src/testing.rs

```diff
@@ -59,26 +66,29 @@ impl Extension {
 
     /// Create a callback handler
     ///
-    /// Returns true if the callback was handled
-    /// returns false if the timeout is reached
+    /// Returns a testing::Result<T,E> if the callback was handled,
+    /// or returns testing::Result::Timeout if the timeout is reached
     ///
-    /// The handler must return a boolean indicating whether the callback was handled
-    /// Return true to end the callback loop
-    /// Return false to continue the callback loop
-    pub fn callback_handler<F>(&self, handler: F, timeout: Duration) -> bool
+    /// The handler must return an enum indicating whether the callback was handled
+    /// Return Result::Ok or Err to end the callback loop
+    /// Return Result::continue to continue the callback loop
+    pub fn callback_handler<F, T, E>(&self, handler: F, timeout: Duration) -> Result<T, E>
     where
-        F: Fn(&str, &str, Option<Value>) -> bool,
+        F: Fn(&str, &str, Option<Value>) -> Result<T, E>,
     {
         let queue = self.callback_queue.clone();
         let start = std::time::Instant::now();
         loop {
             if let Some((name, func, data)) = queue.pop() {
-                if handler(&name, &func, data) {
-                    return true;
+                let option = handler(&name, &func, data);
+                match option {
+                    Result::Ok(value) => return Result::Ok(value),
+                    Result::Err(error) => return Result::Err(error),
+                    _ => (),
```

Comment:
Since you used the same Result for `callback_handler` and the handler function `F`, we need to handle the case in which the handler returns `Result::Timeout`. It should return instead of continuing the loop.